src/components/NameValueInput/NameValueInput.tsx

```diff
@@ -1,39 +1,54 @@
 import React, { useRef } from 'react';
-import { useFieldArray, useFormContext } from 'react-hook-form';
-import { Button, Field, HorizontalGroup, Icon, IconButton, Input, useTheme, VerticalGroup } from '@grafana/ui';
+import { FieldErrorsImpl, useFieldArray, useFormContext } from 'react-hook-form';
+import { Button, Field, HorizontalGroup, Icon, IconButton, Input, useTheme2, VerticalGroup } from '@grafana/ui';
 import { css } from '@emotion/css';
 
+import { CheckFormValues, CheckFormValuesHttp } from 'types';
+
+type NameValueName = 'settings.http.headers' | 'settings.http.proxyConnectHeaders' | 'labels';
+
 interface Props {
-  name: string;
-  limit: number;
+  name: NameValueName;
+  limit?: number;
   disabled?: boolean;
   label: string;
   validateName?: (name: string) => string | undefined;
   validateValue?: (value: string) => string | undefined;
 }
 
+function getErrors(errors: FieldErrorsImpl<CheckFormValues>, name: NameValueName) {
+  switch (name) {
+    case 'settings.http.headers': {
+      const cast = errors as FieldErrorsImpl<CheckFormValuesHttp>;
```

Comment:
I was fighting with this for a long time. I don't get why the new union types won't allow me to say `CheckFormValues` _might_ have HTTP on it. Is there a way to say "I have a form that might be of any check type, but if it's HTTP I would like to use the HTTP key"? Casting always feels icky.

@ckbedwell